crates/types/src/v1alpha1/wasmcloud_host_config.rs

```diff
@@ -187,6 +189,36 @@ fn default_nats_leafnode_port() -> u16 {
     7422
 }
 
+#[derive(Deserialize, Serialize, Clone, Debug, JsonSchema)]
+pub struct ConfigMapSource {
+    /// Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+    pub name: String,
+
+    /// Specify whether the ConfigMap must be defined
+    pub optional: Option<bool>,
+}
+
+#[derive(Deserialize, Serialize, Clone, Debug, JsonSchema)]
+pub struct SecretSource {
+    /// Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+    pub name: String,
+
+    /// Specify whether the ConfigMap must be defined
+    pub optional: Option<bool>,
+}
+
+#[derive(Deserialize, Serialize, Clone, Debug, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct CertificateSource {
+    pub config_map_ref: Option<ConfigMapSource>,
+    pub secret_ref: Option<SecretSource>,
+}
+
+#[derive(Deserialize, Serialize, Clone, Debug, JsonSchema)]
+pub struct WasmCloudHostCertificates {
+    pub authorities: Option<Vec<CertificateSource>>,
```

Comment:
I'm ⚖️ on going with
```
certificates:
  authorities:
```

or straight to `certificateAuthorities`

The motivation for going nested is the ability to add client certificates (or any other ) in the future, so `certificates` becomes the place for all-things-certs.

Reply:
I like the nesting, leaving room for future iterations makes sense

Reply:
I'm also torn on the nesting, but if we had different client certs for OCI and OTEL among other things, then we'd need some sort of nested struct anyway